src/operator/tensor/elemwise_unary_op.h

```diff
@@ -388,6 +388,94 @@ void CastCompute(const nnvm::NodeAttrs& attrs,
   });
 }
 
+struct HardSigmoidParam : public dmlc::Parameter<HardSigmoidParam> {
+  real_t alpha;
+  real_t beta;
+  DMLC_DECLARE_PARAMETER(HardSigmoidParam) {
+    DMLC_DECLARE_FIELD(alpha)
+    .set_default(0.2)
+    .describe("Slope of hard sigmoid");
+    DMLC_DECLARE_FIELD(beta)
+    .set_default(0.5)
+    .describe("Bias of hard sigmoid.");
+  }
+};
+
+template<int req>
+struct hard_sigmoid_forward {
```

Comment:
please inherit tunable for fwd and bwd and add a entry here https://github.com/apache/incubator-mxnet/blob/daf821c09b3df581399a564a1b40b0a2e68750c9/src/operator/operator_tune.cc#L347

see https://github.com/apache/incubator-mxnet/blob/master/src/operator/mxnet_op.h#L596

Reply:
@cjolivier01  would know more about this. He is drafting a doc for autotune. I'll merge this for now. Please revisit after you go through that doc.

Reply:
Sure, I actually talked with him yesterday after seeing your comments and those were what he told me, I think if we see more similar functions with this one then we can definitely get that done.